core/src/main/scala/org/apache/spark/status/api/v1/SimpleDateParam.scala

```diff
@@ -42,14 +42,16 @@ private[v1] class SimpleDateParam(val originalValue: String) {
 
 private[v1] object SimpleDateParam {
 
-  val formats: Seq[SimpleDateFormat] = {
+  // `SimpleDateFormat` is not thread-safe
+  val formats = new ThreadLocal[Seq[SimpleDateFormat]]() {
```

Comment:
It also means you keep per-thread state lying around. I know, it's not big, but I don't think it's worth it.